miniwin/miniwin/src/miniwin_ddraw.cpp

```diff
@@ -80,12 +133,10 @@ struct DirectDrawImpl : public IDirectDraw2, public IDirect3D2 {
 	// IDirect3D2 interface
 	HRESULT CreateDevice(const GUID& guid, void* pBackBuffer, IDirect3DDevice2** ppDirect3DDevice) override
 	{
-		return DDERR_GENERIC;
+		*ppDirect3DDevice = new IDirect3DDevice2;
```

Comment:
I think we need to create a private `miniwin_private.h` header where we put all class declarations, so ddraw can see the classes of d3d and vice versa.

```suggestion
		*ppDirect3DDevice = static_cast<IDirect3DDevice2 *>(new Direct3DDeviceImpl);
```

Reply:
This class has no methods it's basically just a fancy void pointer, no need to cast or implement.

Reply:
True. It's probably used to make sure the device is not accidentally cleaned up.
Something to remember.